### PR TITLE
Russell -L122.ghg_agr_R_S_T_Y

### PR DESCRIPTION
**Pull request by russellhz**
#485

I still have X years in L122.EmissShare_R_C_Y_GLU. Here is what the OLD data system output looks like - I wasn't able to figure out flags to work through this.

screen shot 2017-07-03 at 10 01 11 am

### CODE REVIEW

File: inst/extdata/emissions/EDGAR/EDGAR_nation.csv [hunk 1]
```diff
@@ -0,0 +1 @@
...
asm,ASM
atf,ATF
cck,CCK
gum,GUM
hmd,HMD
mnp,MNP
nfk,NFK
pcn,PCN
tkl,TKL
tuv,TUV
```
Comment (cahartin) [inline inst/extdata/emissions/EDGAR/EDGAR_nation.csv]: Take a look at the wiki, Kate replaced EDGAR nation with 3 lines of code so we don't need this file.

**Comment by bpbond** [comment]
@russellhz - re `L122.EmissShare_R_C_Y_GLU`, ah, that's crappy. We don't have a flag that will make that work. I guess I'll make a new flag just for this.

File: R/zchunk_L122.ghg_agr_R_S_T_Y.R [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+             FILE = "emissions/EDGAR/EDGAR_N2O",
+             FILE = "emissions/EDGAR/EDGAR_NH3",
+             FILE = "emissions/EDGAR/EDGAR_NOx"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L122.EmissShare_R_C_Y_GLU",
+             "L122.ghg_tg_R_agr_C_Y_GLU"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
```
Comment (bpbond) [inline R/zchunk_L122.ghg_agr_R_S_T_Y.R]: Silence package note checks

File: R/zchunk_L122.ghg_agr_R_S_T_Y.R [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    # Emissions shares: product of region/crop shares and region/crop/glu shares
+    L122.EmissShare_R_C_Y_GLU <- L122.ProdGLUshare_R_C_Y_GLU %>%
+      left_join_error_no_match(L122.CropAreaShare_R_C_Y, by = c("GCAM_region_ID", "GCAM_commodity", "year")) %>%
+      transmute(GCAM_region_ID, GCAM_commodity, year, GLU, emiss_share = prod_share_GLU * crop_area_share)
+
+    # Compute EDGAR emissions by region
+    L122.EDGAR <- bind_rows(EDGAR_CH4, EDGAR_N2O, EDGAR_NH3, EDGAR_NOx) %>%
+      # Peat fire and forest fire post burn decay not in EDGAR_sector
+      left_join(EDGAR_sector %>% select(IPCC, sector = agg_sector), by = "IPCC") %>%
```
Comment (bpbond) [inline R/zchunk_L122.ghg_agr_R_S_T_Y.R]: Why not left_join_error_no_match? Do we expect NAs or multiple matches? Clarifying the line 84 comment might help.

File: R/zchunk_L122.ghg_agr_R_S_T_Y.R [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      standardize_iso(col = "ISO_A3") %>%
+      change_iso_code('rou', 'rom') %>%
+      # umi, bvt, sgs, iot, atf, and hmd not in iso_GCAM_regID
```
Comment (bpbond) [inline R/zchunk_L122.ghg_agr_R_S_T_Y.R]: `# Use a left_join because ...`

File: R/zchunk_L122.ghg_agr_R_S_T_Y.R [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+    # Compute emissions from fertilizer by GCAM region, commodity, and GLU
+    L122.EDGAR_fert <- L122.EDGAR_agr %>%
+      filter(sector == "fertilizer")
+
+    #Compute fertilizer by crop
+    L122.fert_Mt_R_C_Y_GLU <- L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      filter(year %in% emissions.EDGAR_YEARS) %>%
+      rename(ag_production = value) %>%
+      left_join_error_no_match(L142.ag_Fert_IO_R_C_Y_GLU %>%
+                                 filter(year %in% emissions.EDGAR_YEARS), ., # Order flipped to deal with timeshift
```
Comment (bpbond) [inline R/zchunk_L122.ghg_agr_R_S_T_Y.R]: Oh, interesting.

**Comment by bpbond** [comment]
Holding on @cahartin approval.

**Pull request merged**